TestFoundation/TestOperationQueue.swift

```diff
@@ -168,34 +168,42 @@ class TestOperationQueue : XCTestCase {
 class AsyncOperation: Operation {
 
     private let queue = DispatchQueue(label: "async.operation.queue")
-    private let lock = NSLock()
+    private let lock = NSRecursiveLock()
```

Comment:
TBH I don't actually like the way `didChangeValue` is written, but I was just doing the simplest fix to get `TestFoundation` to run under Release mode in Xcode, so that it can be run under Instruments, and this was one of the failing tests. Is that 140ns in `-Onone` or `-O`?

If you have an alternate solution I have no problem with using it, this is just test code anyway.